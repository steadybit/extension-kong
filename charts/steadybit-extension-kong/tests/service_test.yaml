templates:
  - service.yaml
tests:
  - it: manifest should match snapshot without TLS
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with mutual TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
        client:
          certificates:
            fromSecrets:
              - client-cert-a
              - client-cert-a
    asserts:
      - matchSnapshot: {}
  - it: manifest should match snapshot with mutual TLS using containerPaths
    set:
      tls:
        server:
          certificate:
            path: /etc/tls/server.crt
            key:
              path: /etc/tls/server.key
        client:
          certificates:
            paths:
              - /etc/tls/ca.crt
              - /etc/tls/ca2.crt
    asserts:
      - matchSnapshot: {}
