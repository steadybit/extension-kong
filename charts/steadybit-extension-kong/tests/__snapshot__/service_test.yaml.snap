manifest should match snapshot with TLS:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        steadybit.com/extension-auto-discovery: "{\n  \"extensions\": [\n    {\n      \"port\": 8084,\n      \"types\": [\"ACTION\",\"DISCOVERY\"],\n      \"tls\": {\n        \"server\": {\n          \"extraCertsFile\": \"/opt/steadybit/agent/etc/extension-mtls/server-cert/tls.crt\"\n        }\n        \n      }\n    }\n  ]\n}\n"
      labels: null
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8084
          protocol: TCP
          targetPort: 8084
      selector:
        app.kubernetes.io/name: steadybit-extension-kong
manifest should match snapshot with mutual TLS:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        steadybit.com/extension-auto-discovery: |
          {
            "extensions": [
              {
                "port": 8084,
                "types": ["ACTION","DISCOVERY"],
                "tls": {
                  "server": {
                    "extraCertsFile": "/opt/steadybit/agent/etc/extension-mtls/server-cert/tls.crt"
                  },
                  "client": {
                    "certChainFile": "/opt/steadybit/agent/etc/extension-mtls/client-cert-a/tls.crt",
                    "certKeyFile": "/opt/steadybit/agent/etc/extension-mtls/client-cert-a/tls.key"
                  }
                }
              }
            ]
          }
      labels: null
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8084
          protocol: TCP
          targetPort: 8084
      selector:
        app.kubernetes.io/name: steadybit-extension-kong
manifest should match snapshot without TLS:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        steadybit.com/extension-auto-discovery: "{\n  \"extensions\": [\n    {\n      \"port\": 8084,\n      \"types\": [\"ACTION\",\"DISCOVERY\"],\n      \"tls\": {\n        \n      }\n    }\n  ]\n}\n"
      labels: null
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8084
          protocol: TCP
          targetPort: 8084
      selector:
        app.kubernetes.io/name: steadybit-extension-kong
